src/association.js

```diff
@@ -0,0 +1,215 @@
+const Utils=require('./utils');
+const Generalization=require('./generalization');
+const Required=require('./required');
+const constant =require('./constant');
+/**
+ *
+ *
+ * @class Association
+/**
+ * Association class returns the API Association 
+ *
+ * @class Association
+ */
+class Association {
+     /**
+      * Creates an instance of Association.
+      * 
+      * @constructor Association
+      */
+     constructor() {
+          this.utils = new Utils();   
+          this.arrAssoc = [];  
+          this.required = new Required();
+     }
+
+     getAssociations() {
+          return this.arrAssoc;
+     }
+     
+     addAssociationProperties(assocClassLink, mainPropertiesObj) {
+          /**
+           * Add asscociation class Properties
+           * eg.
+           *   TransportMeansParty
+                    allOf:
+                    - $ref: '#/components/schemas/TransportPartyIds'
+                    - $ref: '#/components/schemas/TransportMeansParty'
+                    - type: object
+               */
+          if (assocClassLink.length > 0) {
+               assocClassLink.forEach(item => {
+                    this.writeAssociationClassProperties(mainPropertiesObj, item);
+                    this.arrAssoc.push(item.classSide);
+               })
+          }
+          return mainPropertiesObj;
+     }
+     /**
+      * @function writeAssociationClassProperties
+      * @description adds property for association class
+      * @param {Object} main properties json object
+      * @param {UMLAssociationClassLink} associationClass 
+      */
+     writeAssociationClassProperties(mainPropertiesObj, associationClass) {
+          try {
```

Comment:
why do you have these big try blocks? what explicitly are you expecting to fail?